### PR TITLE
WIP: Remove DCL in favor of Component Resolver/ViewContainerRef

### PR DESCRIPTION
**Pull request by brandonroberts**

### CODE REVIEW

File: lib/component-renderer.ts [hunk 1]
```diff
@@ -44,9 +44,9 @@ export class ComponentRenderer {
   render(
     route: Route,
     components: BaseRoute,
-    injector: Injector,
+    injector: ReflectiveInjector,
     ref: ViewContainerRef,
-    dcl: DynamicComponentLoader,
+    compiler: ComponentResolver,
```
Comment (MikeRyan52) [inline lib/component-renderer.ts]: Can this service just inject `ComponentResolver` directly?

File: lib/component-renderer.ts [hunk 2]
```diff
@@ -44,9 +44,9 @@ export class ComponentRenderer {
   render(
     route: Route,
     components: BaseRoute,
-    injector: Injector,
+    injector: ReflectiveInjector,
```
Comment (MikeRyan52) [inline lib/component-renderer.ts]: Change this to just `Injector`

File: lib/component-renderer.ts [hunk 3]
```diff
@@ -56,10 +56,11 @@ export class ComponentRenderer {
       })
       .let(composeHooks(this._preRenderHooks))
       .mergeMap(instruction => {
-        const providers = ReflectiveInjector.resolve(instruction.providers);
+        const instructionInjector = injector.resolveAndCreateChild(instruction.providers);
```
Comment (MikeRyan52) [inline lib/component-renderer.ts]: Change this to:

```ts
ReflectorInjector.resolveAndCreate(providers, injector);
```

File: lib/route-view.ts [hunk 4]
```diff
@@ -49,9 +50,9 @@ export class RouteView implements OnDestroy, OnInit {
   constructor(
     @Attribute('name') private _name: string,
     protected _routerInstruction$: RouterInstruction,
-    protected _injector: Injector,
+    @Inject(Injector) protected _injector: ReflectiveInjector,
```
Comment (MikeRyan52) [inline lib/route-view.ts]: I don't think we need to require a `ReflectiveInjector` here

**Comment by MikeRyan52** [comment]
This is looking good to me 👍 Let me know when it is ready

**Comment by brandonroberts** [comment]
Issues are fixed

**Pull request merged**